From 5d4ec1f1279dc431343def88c95fc5315e30bffc Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Tue, 7 Feb 2017 23:21:29 +0000
Subject: [PATCH] If we are reading a sector stream that is shorter than the
 minimum standard stream size, but there is no short sector chain, adjust the
 dirlen.

Upstream-commit: 781a5357d0dc780e57cc36ec93e47ed3a954b74b
Signed-off-by: Kamil Dudka <kdudka@redhat.com>
---
 src/cdf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cdf.c b/src/cdf.c
index d38e793..b6fe6cd 100644
--- a/src/cdf.c
+++ b/src/cdf.c
@@ -527,7 +527,7 @@ cdf_read_long_sector_chain(const cdf_info_t *info, const cdf_header_t *h,
 	ssize_t nr;
 	scn->sst_tab = NULL;
 	scn->sst_len = cdf_count_chain(sat, sid, ss);
-	scn->sst_dirlen = len;
+	scn->sst_dirlen = MAX(h->h_min_size_standard_stream, len);
 	scn->sst_ss = ss;
 
 	if (scn->sst_len == (size_t)-1)
-- 
2.7.4

